### PR TITLE
Fingerprint Conversion

### PR DESCRIPTION
**Pull request by chickenfromouterspace**
The fingerprint version of conversion.

### CODE REVIEW

File: Arduino/SensorLogic/SensorLogic.ino [hunk 1]
```diff
@@ -115,8 +115,8 @@ void RFIDloop()
     Serial.println("Card detected:");
     for(int i=0;i<5;i++)
     {
-    Serial.print(RC522.serNum[i],DEC);
-    //Serial.print(RC522.serNum[i],HEX); //to print card detail in Hexa Decimal format
+    //Serial.print(RC522.serNum[i],DEC);
+    Serial.print(RC522.serNum[i],HEX); //to print card detail in Hexa Decimal format
```
Comment (chickenfromouterspace) [inline Arduino/SensorLogic/SensorLogic.ino:119]: This change didn't really matter. It only made the string look shorter.

File: Arduino/libraries/Capacitive_Fingerprint_Reader/finger.cpp [hunk 2]
```diff
@@ -9,16 +9,12 @@ uint8_t finger_TxBuf[9];
...
-SoftwareSerial mySerial(10, 11); // RX, TX		
-
-
 /***************************************************************************
 * @brief      initialize the SoftwareSerial to communicate with Fingerprint module
 ****************************************************************************/
 void Finger_SoftwareSerial_Init(void)
 {
-	mySerial.begin(19200);	
+	Serial2.begin(19200);
```
Comment (chickenfromouterspace) [inline Arduino/libraries/Capacitive_Fingerprint_Reader/finger.cpp:17]: Serial2 refers to TX2/RX2. Same applies for Serial1 and Serial3 for TX1/RX1 and TX3/RX3 respectively.

File: Arduino/libraries/Capacitive_Fingerprint_Reader/finger.cpp [hunk 3]
```diff
@@ -58,16 +54,16 @@ uint8_t TxAndRxCmd(uint8_t Scnt, uint8_t Rcnt, uint16_t Nms)
 	 
 	 memset(finger_RxBuf,0,sizeof(finger_RxBuf));   ////////
  
-	 mySerial.flush();  /////
+	 serialFlush();  /////
```
Comment (chickenfromouterspace) [inline Arduino/libraries/Capacitive_Fingerprint_Reader/finger.cpp:57]: Had to create a new serialFlush function due to how Serial2.flush() worked. Currently Serial2.flush() waits for serial transmission to complete. We need the version that clears the serial buffer.

File: Arduino/libraries/Capacitive_Fingerprint_Reader/finger.h [hunk 4]
```diff
@@ -66,6 +66,7 @@ uint8_t GetTimeOut(void);
 void Finger_Wait_Until_OK(void);	
 void Analysis_PC_Command(void);
 void Auto_Verify_Finger(void);
+void serialFlush(void);
```
Comment (chickenfromouterspace) [inline Arduino/libraries/Capacitive_Fingerprint_Reader/finger.h:69]: Constructor for the new function.

File: Arduino/libraries/Capacitive_Fingerprint_Reader/finger.cpp [hunk 5]
```diff
@@ -473,4 +469,10 @@ void Auto_Verify_Finger(void)
 			digitalWrite(Finger_RST_Pin , LOW);    
 		}
 	}
-}
\ No newline at end of file
+}
+
+void serialFlush(){
```
Comment (chickenfromouterspace) [inline Arduino/libraries/Capacitive_Fingerprint_Reader/finger.cpp]: Basically if there is anything left in the Serial2 buffer, it places it inside a character that does absolutely nothing. Serial2.read() automatically pushes the string along so doing this until Serial2 is unavailable is the same as emptying out the serial buffer.

**Review by chickenfromouterspace — COMMENTED** [review]
These are my findings in regards to converting the fingerprint.

File: Arduino/libraries/Capacitive_Fingerprint_Reader/finger.cpp [hunk 6]
```diff
@@ -473,4 +469,10 @@ void Auto_Verify_Finger(void)
...
+  while(Serial2.available() > 0) {
+    char t = Serial2.read();
+  }
```
Comment (chickenfromouterspace) [inline Arduino/libraries/Capacitive_Fingerprint_Reader/finger.cpp:477]: Basically if there is anything left in the Serial2 buffer, it places it inside a character that does absolutely nothing. Serial2.read() automatically pushes the string along so doing this until Serial2 is unavailable is the same as emptying out the serial buffer.

**Review by buffalosoldier386 — APPROVED** [review]
Good job.

**Pull request merged**